### DISCUSSION TITLE
IE11 support

### DISCUSSION THREAD

**Discussion by dmethvin-gov** [CMSgov/design-system#2845] [General]
The [browser support](https://design.cms.gov/guidelines/browser-support/?query=browser&theme=core) page says CMSDS doesn't support IE11, which makes sense. I've worked on several sites/apps with CMSDS and none have ever tested or claimed support for IE11. The current US browser market share stats for IE11 show it well below 1 percent. 

However, the project still has [several polyfills and a few stray mentions](https://github.com/search?q=repo%3ACMSgov%2Fdesign-system%20ie11&type=code) of IE11. The main negative impact is the polyfills, they seem to add about 100KB to the bundle due to core-js. As far as I can tell, nothing else in our project uses core-js so this is all from CMSDS.

<img width="694" alt="Screenshot 2023-12-20 at 10 30 44 AM" src="https://github.com/CMSgov/design-system/assets/40182224/5c8b59a9-db36-4ba3-9cb9-9d3cbb96154d">

The only counterargument I can think of for leaving IE11 support is that [the 2021 WebAIM survey] said about 3 percent of users of assistive tech (mainly JAWS I'd think) are still using IE11. However, since neither the design system nor the apps built on it are testing with IE11, the chances are zero that they work with IE11, because [Untested Code is Broken Code](https://www.redbubble.com/i/t-shirt/Untested-Code-is-Broken-Code-Tattoo-by-gsansone4/33696986.NL9AC). Anyone who claims to need IE11 support should prove that their own project actually tests with it, and in any case could include the polyfills themselves.

So would it be reasonable to remove those polyfills (plus the mention of IE11+JAWS and the one CSS rule)?

**Comment by pwolfert** [comment]
Thanks for bringing this up, @dmethvin-gov. I removed them locally and found that it saved 38KB uncompressed in our bundles (see notes below). That's not nothing, though I was hoping it would match what your bundle analyzer predicted. I ran a bundle analyzer on ours before and after, and it too reported `core-js` taking up around 100KB. In actuality, it was lower, and I did verify that core-js was no longer in the bundle. We'll discuss adding this change to v9, which will be released in late January. Thanks!

```
# Before
react-components.js     575KB
preact-components.js    593KB

# After
react-components.js     537KB
preact-components.js    555KB

# Diff
react-components.js      38KB (6.6%)
preact-components.js     38KB (6.4%)
```

Before (includes core-js):
<img width="414" alt="Bundle analyzer showing the core-js library in or bundle" src="https://github.com/CMSgov/design-system/assets/7595652/6e4e5b36-4d2f-435d-809d-d1583ac1512f">

After (does not include core-js):
<img width="1475" alt="Full bundle analysis showing that core-js is no longer in the bundle" src="https://github.com/CMSgov/design-system/assets/7595652/1898a271-a71f-4f03-9d00-442212ab1fb1">

**Comment by dmethvin-gov** [comment]
That's super strange! At least it did drop down a bit. That first shot shows zero bytes for core-js, which is also strange. Thanks for looking into it! I agree that it should wait for a major version.

**Comment by pwolfert** [comment]
Addressed by https://github.com/CMSgov/design-system/pull/2866
